core/src/test/java/org/mule/expression/ExpressionManagerTestCase.java

```diff
@@ -141,6 +148,124 @@ public void testBooleanEvaluation()
         assertFalse(muleContext.getExpressionManager().evaluateBoolean("string:false", msg));
     }
 
+    @Test
+    public void testDefaultExpressionFilterActsLikeBooleanValueOfInvalidStringUsingSystemProperty() throws Exception
+    {
+        testWithSystemProperty(MULE_EXPRESSION_FILTER_DEFAULT_BOOLEAN_VALUE, "false", new TestCallback()
+        {
+            @Override
+            public void run()
+            {
+                ExpressionFilter filter = new ExpressionFilter("payload");
+                assertThat(filter.isNonBooleanReturnsTrue(), is(false));
+
+                filter.setMuleContext(muleContext);
+
+                assertThat(filter.accept(new DefaultMuleMessage("on", muleContext)), is(valueOf("on")));
```

Comment:
All these assertions are always repeated. Extract the logic to a reusable function.